test/common/stats/symbol_table_fuzz_test.cc

```diff
@@ -69,6 +44,51 @@ DEFINE_FUZZER(const uint8_t* buf, size_t len) {
       }
       TestUtil::serializeDeserializeNumber(number);
     }
+
+    // We can add stat-names with trailing dots, but note that they will be
+    // trimmed by the Symbol Table implementation, so we must trim the input
+    // string before comparing.
+    absl::string_view trimmed_fuzz_data = StringUtil::removeTrailingCharacters(next_data, '.');
+    FUZZ_ASSERT(trimmed_fuzz_data == symbol_table.toString(stat_name));
+    FUZZ_ASSERT(trimmed_fuzz_data == fake_symbol_table.toString(fake_stat_name));
+    FUZZ_ASSERT(trimmed_fuzz_data == symbol_table.toString(dynamic_stat_name));
+    FUZZ_ASSERT(trimmed_fuzz_data == fake_symbol_table.toString(fake_dynamic_stat_name));
+
+    // The 'join' tests only work if the trimmed fuzz data is not empty.
+    if (trimmed_fuzz_data.empty()) {
+      continue;
+    }
+
+    std::string joined = absl::StrCat(trimmed_fuzz_data, ".", trimmed_fuzz_data);
+    auto join = [joined](SymbolTable& table, StatName name1, StatName name2) -> bool {
+      bool ok = true;
+      SymbolTable::StoragePtr storage = table.join({name1, name2});
+      StatName stat_name(storage.get());
+      std::string name1_name2 = table.toString(stat_name);
+      if (joined.size() != name1_name2.size()) {
+        std::cerr << "lengths don't match: " << joined.size() << " != " << name1_name2.size()
+                  << std::endl;
+        ok = false;
+      } else {
+        for (uint32_t i = 0; i < joined.size(); ++i) {
+          if (joined[i] != name1_name2[i]) {
+            std::cerr << "char [" << i << "] mismatch: " << joined[i] << "("
```

Comment:
Is there a reason to keep these as `std::cerr` over `ENVOY_LOG`? (Maybe you used them to print when running the fuzzer? I haven't been checking in these because printing log statements will slow down the fuzzer in OSS-Fuzz).

Reply:
good call; done.